### PR TITLE
Attempted fix for eventual Vulkan errors.

### PR DESCRIPTION
**Pull request by ScottTodd**
Fixes https://github.com/google/iree/issues/3381 (or at least appears to for me)

### CODE REVIEW

File: bindings/python/pyiree/rt/function_abi.cc [hunk 1]
```diff
@@ -833,10 +833,13 @@ void SetupFunctionAbiBindings(pybind11::module m) {
...
-        return self->Unpack(absl::MakeConstSpan(self->raw_config().results),
-                            f_results);
-      });
+      .def(
+          "unpack_results",
+          [](FunctionAbi* self, VmVariantList& f_results) {
+            return self->Unpack(absl::MakeConstSpan(self->raw_config().results),
+                                f_results);
+          },
+          py::keep_alive<0, 1>());
```
Comment (stellaraccident) [inline bindings/python/pyiree/rt/function_abi.cc:842]: Can you add a TODO that this simple keep_alive mechanism is unlikely to work for multi-return functions?

**Review by stellaraccident — APPROVED** [review]

File: bindings/python/pyiree/rt/function_abi.cc (hunk 1 above)
Comment (ScottTodd) [reply bindings/python/pyiree/rt/function_abi.cc:844]: Sure, added TODO.

File: bindings/python/pyiree/rt/function_abi.cc [hunk 2]
```diff
@@ -833,10 +833,15 @@ void SetupFunctionAbiBindings(pybind11::module m) {
...
-      });
+      // TODO(laurenzo): Generalize this keep_alive fix (for #3381) to handle
+      //                 multi-return functions
+      .def(
+          "unpack_results",
+          [](FunctionAbi* self, VmVariantList& f_results) {
+            return self->Unpack(absl::MakeConstSpan(self->raw_config().results),
+                                f_results);
+          },
+          py::keep_alive<0, 1>());
```
Comment (ScottTodd) [inline bindings/python/pyiree/rt/function_abi.cc:844]: Some test failures:

```
ERROR: test_add_scalar (__main__.VmTest)
VmTest.test_add_scalar
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmpfs/src/github/iree/bindings/python/pyiree/rt/vm_test.py", line 138, in test_add_scalar
    result = abi.unpack_results(allocated_results)
RuntimeError: Could not allocate weak reference!

----------------------------------------------------------------------
Ran 9 tests in 2.937s

FAILED (errors=1)
```

Logs: https://source.cloud.google.com/results/invocations/4cb4b8be-635f-4928-a098-567ee13c5cb4/targets/iree%2Fgcp_ubuntu%2Fcmake%2Flinux%2Fx86-swiftshader%2Fpresubmit/log

**Comment by ScottTodd** [comment]
This didn't work, and the pyiree code it affected is obsolete now.

**Pull request closed without merging**